include/fastdds/rtps/reader/RTPSReader.h

```diff
@@ -251,6 +252,12 @@ class RTPSReader
         return mp_history;
     }
 
+    RTPS_DllAPI void set_content_filter(
+            eprosima::fastdds::rtps::IReaderDataFilter* filter)
+    {
```

Comment:
Maybe we can lock the mutex here. What'd happen if the user sets a `nullptr` filter while we are operating with it?